File: app/src/main/java/com/fpoly/project1/activity/authentication/AuthResetPassword.java
package com.fpoly.project1.activity.authentication;

import android.content.Intent;
import android.os.Bundle;
import android.widget.EditText;
import android.widget.Toast;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.fpoly.project1.R;
import com.fpoly.project1.activity.MainActivity;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;

public class AuthResetPassword extends AppCompatActivity {
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_reset_password);

        EditText inputPassword = findViewById(R.id.resetPassword_edt_newPass);
        EditText inputPasswordConfirm = findViewById(R.id.resetPassword_edt_confirmNewPass);

        findViewById(R.id.resetPassword_btn_save).setOnClickListener(v -> {
            boolean hasError = false;

            if (inputPassword.getText().toString().length() == 0) {
                inputPassword.setError("Field must not be empty");
                hasError = true;
            }
            if (inputPasswordConfirm.getText().toString().length() == 0) {
                inputPasswordConfirm.setError("Field must not be empty");
                hasError = true;
            }
            if (!inputPassword.getText().toString().equals(inputPasswordConfirm.getText().toString())) {
                inputPassword.setError("Password mismatch");
                inputPasswordConfirm.setError("Password mismatch");
                hasError = true;
            }
            if (!hasError) {
                FirebaseUser user = FirebaseAuth.getInstance().getCurrentUser();

                if (user != null) {
                    user.updatePassword(inputPassword.getText().toString())
                            .addOnCompleteListener(task -> {
                                if (task.isSuccessful()) {
                                    Toast.makeText(AuthResetPassword.this, "Successfully reset password", Toast.LENGTH_SHORT).show();

                                    startActivity(new Intent(AuthResetPassword.this, MainActivity.class));
                                } else {
                                    Toast.makeText(AuthResetPassword.this, "Failed to set new password", Toast.LENGTH_SHORT).show();
                                }
                            });
                } else {
                    Toast.makeText(AuthResetPassword.this, "Failed to get session", Toast.LENGTH_SHORT).show();
                }
            }
        });
    }
}

File: app/src/main/java/com/fpoly/project1/activity/authentication/AuthResetPassword.kt
package com.fpoly.project1.activity.authentication

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.fpoly.project1.R
import com.google.firebase.auth.FirebaseAuth

class AuthResetPassword : AppCompatActivity() {
    private lateinit var etPassword: EditText
    private lateinit var etPasswordConfirm: EditText

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.account_change_pwd)

        etPassword = findViewById(R.id.changePassword_edt_newPass)
        etPasswordConfirm = findViewById(R.id.changePassword_edt_confirmNewPass)

        findViewById<Button>(R.id.changePassword_btn_save).setOnClickListener {
            var hasError = false
            if (etPassword.text.toString().isEmpty()) {
                etPassword.error = "Field must not be empty"
                hasError = true
            }
            if (
                etPasswordConfirm.text.toString().isEmpty()
            ) {
                etPasswordConfirm.error = "Field must not be empty"
                hasError = true
            }
            if (etPassword.text.toString() != etPasswordConfirm.text.toString()) {
                etPassword.error = "Password mismatch"
                etPasswordConfirm.error = "Password mismatch"
                hasError = true
            }
            if (!hasError) {
                val user = FirebaseAuth.getInstance().currentUser
                user?.updatePassword(etPassword.text.toString())
                    ?.addOnCompleteListener { task ->
                        if (task.isSuccessful) {
                            Toast.makeText(
                                this@AuthResetPassword,
                                "Successfully reset password",
                                Toast.LENGTH_SHORT
                            ).show()
                            startActivity(
                                Intent(
                                    this@AuthResetPassword,
                                    AuthLogin::class.java
                                )
                            )
                            finish()
                        } else {
                            Toast.makeText(
                                this@AuthResetPassword,
                                "Failed to set new password",
                                Toast.LENGTH_SHORT
                            ).show()
                        }
                    }
                    ?: Toast.makeText(
                        this@AuthResetPassword,
                        "Failed to get session",
                        Toast.LENGTH_SHORT
                    ).show()
            }
        }
    }
}
